src/schemas/map.json

```diff
@@ -35,9 +35,39 @@
       "type": "number",
       "minimum": 0
     },
+    "MaxConcurrencyPath": {
+      "$ref": "paths.json#/definitions/asl_ref_path"
+    },
     "ItemReader": {
-      "$comment": "Allow any object for now and update in a subsequent PR to support S3 buckets and other input sources",
-      "type": "object"
+      "$comment": "A Map State MAY have an \"ItemReader\" field, whose value MUST be a JSON object and is called the ItemReader Configuration",
+      "type": "object",
```

Comment:
The ItemReader changes went in quickly a few months ago

circling back now to address this comment and flesh out the schema a bit more with the types as documented in the states language.

I included snippets of the schema text here near the relevant rules